src/main/resources/org/jenkinsci/plugins/workflow/job/WorkflowRun/sidepanel.jelly

```diff
@@ -31,7 +31,17 @@
             <l:task icon="images/24x24/up.png" href="${rootURL}/${it.parent.url}" title="${%Back to Project}" contextMenu="false"/>
             <l:task icon="images/24x24/search.png" href="${buildUrl.baseUrl}/" title="${%Status}" contextMenu="false"/>
             <l:task icon="images/24x24/notepad.png" href="${buildUrl.baseUrl}/changes" title="${%Changes}"/>
-            <p:console-link/>
+            <j:choose> <!-- TODO <p:console-link/> may not currently be used as it calls getLogFile -->
+                <j:when test="${it.logText.length() > 200000}">
```

Comment:
Perhaps `LogStorage` needs a new API like

```java
default int approximateLogLines(@Nonnull FlowExecutionOwner.Executable build) {
    return overallLog(build, false).length() / 80;
}
```

which could be overridden by external implementations to make an API call.

Of course there is the broader goal of [JENKINS-17406](https://issues.jenkins-ci.org/browse/JENKINS-17406) to use paginated display.

Reply:
Specifically:

* `console-link.jelly` uses `it.logFile.length()`, which for `WorkflowRun` now prints a warning—very bad. Thus we specifically avoid this control until core is fixed. Fortunately it is not much to copy.
* `console.jelly` uses `it.logText.length()`, which is fine for the built-in `FileLogStorage`: boils down to a single `File.length()` call after verifying that the first few bytes of the file are not a gzip magic header.
* For `pipeline-log-fluentd-cloudwatch` and the like, this is very inefficient, and we would prefer an API for an approximation. (On reflection this need not be in `LogStorage`; could be in `LargeText` or `AnnotatedLargeText`.)

Reply:
Needs to be revisited when resuming work on external log sinks. I am no longer working on that.